-1

Tested at: e7fb44ac9ee5141085dcb4d277fe36f82f1503fc


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/dbd2bf1e094318b1f30ca27f22da19f3ebf61bcf
https://github.com/cms-sw/cmssw/commit/68226b6d643c7b3fdb79c45a5b51a521821f6efc
https://github.com/cms-sw/cmssw/commit/943fb07065e18d9b6d23bb188d6b5a63dc7bceee
https://github.com/cms-sw/cmssw/commit/d73a7d340dec5f711b14e299c65e09f6c64df252
https://github.com/cms-sw/cmssw/commit/8cf418d85b02686ca4208ecf1525ba0843829df4
https://github.com/cms-sw/cmssw/commit/b974b2f5f67d91937ce5067d8ade1f7ce669165a
https://github.com/cms-sw/cmssw/commit/22b26f3c96898ec0e7204f29a5376c9b2b842f5b
https://github.com/cms-sw/cmssw/commit/52c7cbec00d9be17d3f2084900a73154803982ea
https://github.com/cms-sw/cmssw/commit/775ec83a4bd89fd5ef0e1ce03d64c57b42b44e24
https://github.com/cms-sw/cmssw/commit/2b6b83829ffcb7f2010c5d2a7c1e4bcbc5c1dde3
https://github.com/cms-sw/cmssw/commit/be887529af01e390bad207eb18f8b73363ad7406
https://github.com/cms-sw/cmssw/commit/5e53c2b9539aa99597276fc2af455b7acffca3cf
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20595/23102/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20595/23102/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20595/23102/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found an error when building: 
 
 <pre>                 from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-09-19-1100/src/FWCore/Framework/interface/EDProducer.h:12,
                 from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-09-19-1100/src/PhysicsTools/UtilAlgos/interface/StringBasedNTupler.h:8,
                 from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-09-19-1100/src/PhysicsTools/UtilAlgos/src/StringBasedNTupler.cc:1:
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-09-19-1100/src/FWCore/Framework/interface/ProductRegistryHelper.h: In instantiation of 'edm::ProductRegistryHelper::BranchAliasSetter& edm::ProductRegistryHelper::BranchAliasSetterT<T>::setBranchAlias(std::__cxx11::string) [with T = unsigned int; std::__cxx11::string = std::__cxx11::basic_string<char>]':
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-09-19-1100/src/PhysicsTools/UtilAlgos/interface/StringBasedNTupler.h:357:62:   required from here
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-09-19-1100/src/FWCore/Framework/interface/ProductRegistryHelper.h:62:16: error: invalid initialization of non-const reference of type 'edm::ProductRegistryHelper::BranchAliasSetter&' from an rvalue of type 'edm::ProductRegistryHelper::BranchAliasSetterT<unsigned int>*'
         return this;
                ^~~~
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-09-19-1100/src/FWCore/Framework/interface/ProductRegistryHelper.h: In instantiation of 'edm::ProductRegistryHelper::BranchAliasSetter& edm::ProductRegistryHelper::BranchAliasSetterT<T>::setBranchAlias(std::__cxx11::string) [with T = std::vector<float>; std::__cxx11::string = std::__cxx11::basic_string<char>]':
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-09-19-1100/src/PhysicsTools/UtilAlgos/interface/StringBasedNTupler.h:363:90:   required from here
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_4_X_2017-09-19-1100/src/FWCore/Framework/interface/ProductRegistryHelper.h:62:16: error: invalid initialization of non-const reference of type 'edm::ProductRegistryHelper::BranchAliasSetter&' from an rvalue of type 'edm::ProductRegistryHelper::BranchAliasSetterT<std::vector<float> >*'
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/dbd2bf1e094318b1f30ca27f22da19f3ebf61bcf
https://github.com/cms-sw/cmssw/commit/68226b6d643c7b3fdb79c45a5b51a521821f6efc
https://github.com/cms-sw/cmssw/commit/943fb07065e18d9b6d23bb188d6b5a63dc7bceee
https://github.com/cms-sw/cmssw/commit/d73a7d340dec5f711b14e299c65e09f6c64df252
https://github.com/cms-sw/cmssw/commit/8cf418d85b02686ca4208ecf1525ba0843829df4
https://github.com/cms-sw/cmssw/commit/b974b2f5f67d91937ce5067d8ade1f7ce669165a
https://github.com/cms-sw/cmssw/commit/22b26f3c96898ec0e7204f29a5376c9b2b842f5b
https://github.com/cms-sw/cmssw/commit/52c7cbec00d9be17d3f2084900a73154803982ea
https://github.com/cms-sw/cmssw/commit/775ec83a4bd89fd5ef0e1ce03d64c57b42b44e24
https://github.com/cms-sw/cmssw/commit/2b6b83829ffcb7f2010c5d2a7c1e4bcbc5c1dde3
https://github.com/cms-sw/cmssw/commit/be887529af01e390bad207eb18f8b73363ad7406
https://github.com/cms-sw/cmssw/commit/5e53c2b9539aa99597276fc2af455b7acffca3cf
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20595/23102/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20595/23102/git-merge-result

--- a/FWCore/Framework/interface/ProductRegistryHelper.h
+++ b/FWCore/Framework/interface/ProductRegistryHelper.h
@@ -57,7 +57,7 @@ namespace edm {
       BranchAliasSetterT( BranchAliasSetter&& iS):
       value_(iS.value_), token_(iS.token_.index()) {}
       
-      BranchAliasSetter& setBranchAlias(std::string alias) {
+      BranchAliasSetterT<T>& setBranchAlias(std::string alias) {
         value_.branchAlias_ = std::move(alias);
         return *this;
       }

--- a/FWCore/Framework/interface/ProductRegistryHelper.h
+++ b/FWCore/Framework/interface/ProductRegistryHelper.h
@@ -59,7 +59,7 @@ namespace edm {
       
       BranchAliasSetter& setBranchAlias(std::string alias) {
         value_.branchAlias_ = std::move(alias);
-        return this;
+        return *this;
       }
       TypeLabelItem& value_;
       EDPutTokenT<T> token_;

